app/services/hmpps_api/complexity_api.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module HmppsApi
+  class ComplexityApi
+    class << self
+      def get_complexity offender_no
+        route = "/complexity-of-need/offender-no/#{offender_no}"
+        begin
+          result = client.get route, cache: false
+          result.fetch('level')
+        rescue Faraday::ResourceNotFound
+          nil
+        end
+      end
+
+      def save offender_no, level:, username:, reason:
+        route = "/complexity-of-need/offender-no/#{offender_no}"
+        client.post route, { level: level, sourceUser: username, notes: reason }, cache: false
+      end
+
+      def get_complexities offender_nos
+        route = '/complexity-of-need/multiple/offender-no'
+        result = client.post route, [offender_nos], cache: false
+        result.map { |complexity| [complexity.fetch('offenderNo'), complexity.fetch('level')] }.to_h
+      end
+
+      def get_history offender_no
+        route = "/complexity-of-need/offender-no/#{offender_no}/history"
+        results = client.get route, cache: false
+        results.sort_by { |item| item.fetch('createdTimeStamp') }.map do |result|
+          {
+          level: result.fetch('level'),
+          createdTimeStamp: DateTime.parse(result.fetch('createdTimeStamp')),
+          sourceUser: result.fetch('sourceUser')
```

Comment:
`sourceUser` is an optional field – it will be omitted from the response if one wasn't given for that record.

To avoid `KeyError (key not found: "sourceUser")` we should default to `nil`:

```suggestion
          sourceUser: result['sourceUser']
```

(or follow the same pattern as the `notes` field...)